### PR TITLE
Nerf aliums

### PR DESCRIPTION
**Pull request by ghost**
:cl: steamp0rt
balance: All xenos, except for queens, can now be stunned. They DO recover much more rapidly than humans, however.
/:cl:

### CODE REVIEW

**Comment by Lxtoofonea** [comment]
-1. xenos are meant to be OP, one hunter is meant to be a good solo antag. you are suppose to catch the nest early, and in my opinion when an admin spawns an alien larva, they want to do something that is going to end the round.

**Comment by ghost** [comment]
@Lxtoofonea Yes, but it might be IMPOSSIBLE to catch early if the larva spawns in an obscure place and immediately ventcrawls

**Comment by ghost** [comment]
+1, was going to do exactly this sometime in the future. Also make them slippable.

**Comment by ghost** [comment]
@HideAndSeekLOGIC this does. although again, they DO recover from mobility damage at an _additional_ rate of 2.5% of total per tick

**Comment by WarRage333** [comment]
+1 maybe now when admemes spawn aliums crew can actually stand a chance, you just straight out can't kill an alien that's decent at the game

**Comment by Manly-Dwarf** [comment]
-1
Xenos are intended as a round ending event for a reason. As a whole, I don't think xenos are very well designed, but this is pure grudgecode and an extremely stupid way to deal with xenos considering you could address things like Flamethrowers not being good against them (as they should be) instead.

As a sidenote, I'm surprised so many people attempt to grudgecode Xenos instead of Security Borgs, which are mostly unstunnable (w/o an ion gun or flashbang) and have near-infinite ammo in their disabler, which can't be disarmed, on top of all-access and being able to bolt doors, basically meaning if one chases you you're fucked. Thankfully Secborg players tend to be unrobust, which might help a lot in ensuring borgs never get discussed.

**Comment by ghost** [comment]
@Kawaii-Big-Boss they're supposed to be an *event*, not a *round-ending* event. Also, this nerf isn't completely game changing so Xenos are still strong.

**Comment by Manly-Dwarf** [comment]
They are deliberately designed to be so overpowered that they will end rounds either through deathsquad or shuttle calls, think something like Wizards or Nuke Teams spawning off of random events, albeit not on par with how powerful xenos can be in some cases.
I've seen many xenos fail miserably because they didn't pick a good starting spot, didn't get enough infections before they died, or just _HUNTRR NOT QWEEEN_
I call this shitty grudgecode because that's what it is. Right now, when pretty much all of the older players can admit that stuns are insanely overpowered or suck balls, tying the game even further into a shitty stun system which nobody likes will do no favors. Also, I brought up Security Borgs, and I can guarantee that a secborg with a disabler, flash or harmbaton could easily wreck all of the xeno's shit without trying assuming they knew how to not let them get close and spam down halls. Not to mention Riot Shotguns and other things as minor as lube and soap make playing as the Hunter a complete gamble, because the speed can very quickly result in you slipping on shit you couldn't see coming and getting blasted by a guy hiding in a locker.

It's the type of design that someone would make them they were mad because they died. Sounds good at first, but when you think about it for more than two seconds, it's stupid and terrible.

**Comment by ghost** [comment]
@Kawaii-Big-Boss that's why I suggested giving them a 'rush' ability in addition to giving them stuns, but that can always be added later. @steamp0rt , can you give me some numbers for how much this actually reduced stun by?

**Comment by ghost** [comment]
@HideAndSeekLOGIC 

Red line is humans, blue line is aliums. X value is time, Y is mobility damage (stun/knockdown/paralysis/immobilized)

capture

**Comment by ghost** [comment]
Note that the graph isn't entirely accurate, as the blue line represents ADDITIONAL immobility regen, not total.

**Comment by ghost** [comment]
capture

**Comment by ghost** [comment]
@steamp0rt how much will, for example, getting stunprodded change for aliums? Also, do you have something like a 3 second stun cap for the aliens?

**Comment by ghost** [comment]
i haven't actually tested how long it'll last

**Comment by ghost** [comment]
And doing math for good ol' Dy/dx = -y^(39/40) - 1 is not something fun.

**Comment by ghost** [comment]
wfhwbzuous

**Comment by ghost** [comment]
tl;dw: they'll regen stamina damage faster than a disabler or taser can dish it, but stunguns/stunbatons will take em out

**Comment by ghost** [comment]
@steamp0rt is it possible to make it so that their mobility dam can't drop below 3 seconds worth? I feel like stunbatons are still too strong against them.

**Comment by ghost** [comment]
@HideAndSeekLOGIC I don't know how mobility damage is calculated, or translates to seconds.

File: code/modules/mob/living/carbon/alien/damage_procs.dm [hunk 1]
```diff
@@ -5,9 +5,10 @@
 /mob/living/carbon/alien/adjustToxLoss(amount, updating_health = TRUE, forced = FALSE) //alien immune to tox damage
 	return FALSE
 
-//aliens are immune to stamina damage.
+/* hippie -- make aliums stunnable
 /mob/living/carbon/alien/adjustStaminaLoss(amount, updating_stamina = 1)
```
Comment (k-k-karma) [inline code/modules/mob/living/carbon/alien/damage_procs.dm]: modularize

File: code/modules/mob/living/carbon/alien/humanoid/caste/hunter.dm [hunk 2]
```diff
@@ -47,7 +47,9 @@
 		leaping = 1
 		weather_immunities += "lava"
 		update_icons()
+		add_trait(TRAIT_STUNIMMUNE, "hippie-alium") // hippie -- make it so you can't stun aliums midair
 		throw_at(A, MAX_ALIEN_LEAP_DIST, 1, src, FALSE, TRUE, callback = CALLBACK(src, .proc/leap_end))
+		remove_trait(TRAIT_STUNIMMUNE, "hippie-alium") // hippie -- make it so you can't stun aliums midair
```
Comment (k-k-karma) [inline code/modules/mob/living/carbon/alien/humanoid/caste/hunter.dm]: modularize, and put the remove_trait in leap end callback

**Comment by ghost** [comment]
agreed this is shit,  make xenos git fucking JITTERY when on fire and also scream continuously and let them slip but only for maximum half a second

**Comment by DaydreamPuppeh** [comment]
Xenos need to be reworked. Not nerfed like this

**Comment by ghost** [comment]
Stun time is now capped at 3 seconds

File: code/modules/mob/living/carbon/alien/damage_procs.dm (hunk 1 above)
Comment (k-k-karma) [reply code/modules/mob/living/carbon/alien/damage_procs.dm]: still modularizable, make it call the carbon proc

**Comment by bardofrage** [comment]
-1, Aliens are meant to be very strong, they're not a very big threat if they can be stun batoned to death just like anyone else. They're not a normal antag and shouldn't be treated as such.

**Comment by ghost** [comment]
@bardofrage Yeah, but it's no fun if they shitstomp the station EVERY. SINGLE. TIME.

**Comment by ghost** [comment]
make grabs actually disable people if you grab an arm it gets unusable,  leg makes u slow,  etc.  that way 5 guys can grab a xeno and rape it

**Comment by earthmoonjune** [comment]
-1

**Comment by ghost** [comment]
@t-f-w Explain why.

**Comment by earthmoonjune** [comment]
every reason why has already been explained above.

**Comment by ghost** [comment]
Yeah but aliens shouldn't be too OP to stop

**Comment by k-k-karma** [comment]
-3 from headmins. Reopen if you rework this to their views

**Comment by ghost** [comment]
yeah but their views are retarded as fuck

**Comment by ghost** [comment]
do they have brain damage or something

**Pull request closed without merging**